code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -207,6 +207,15 @@
 	do_sparks(3, 1, src)
 	..()
 
+///Disassembling the bot in a civilized manner with a multitool
+/mob/living/simple_animal/bot/cleanbot/disassemble()
+	on = FALSE
```

Comment:
Same here, do we need it to be turned off?

Reply:
I removed all `walk_to(src,0)` and `on = FALSE` from the `disassemble()` and noticed no problem while interrupting movement / actions of the bots by disassembling them.